pulsar-impl/src/main/java/com/datastax/oss/sink/pulsar/AvroTypeUtil.java

```diff
@@ -15,24 +15,131 @@
  */
 package com.datastax.oss.sink.pulsar;
 
+import static com.datastax.oss.sink.pulsar.CqlLogicalTypes.CQL_DECIMAL;
+import static com.datastax.oss.sink.pulsar.CqlLogicalTypes.CQL_DURATION;
+import static com.datastax.oss.sink.pulsar.CqlLogicalTypes.CQL_VARINT;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
+import org.apache.avro.Conversion;
+import org.apache.avro.LogicalType;
+import org.apache.avro.LogicalTypes;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.IndexedRecord;
 import org.apache.pulsar.client.api.schema.GenericRecord;
 import org.apache.pulsar.common.schema.SchemaType;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public final class AvroTypeUtil {
 
-  private static final Logger log = LoggerFactory.getLogger(AvroTypeUtil.class);
+  private static Map<String, Conversion<?>> logicalTypeConverters = new HashMap<>();
+  private static boolean decodeCDCDataTypes;
```

Comment:
we need something to reset this value.

In production there is no big problem, because every Sink runs into its own Classloader, so multiple Sinks inside the same JVM are isolated.

but probably in unit tests this is a problem

Reply:
good point. will do. the only caveat is I cannot deregister the Avro types on org.apache.avro.LogicalTypes because there is no "deregister api" - but I still have control over the flag on our side...